File: banking-ui/src/app/service/account.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Account {
  id?: number;
  accountNumber: string;
  accountType: string;
  balance: number;
  customerId: number;
  status?: string;
  createdDate?: string;
}

@Injectable({ providedIn: 'root' })
export class AccountService {
  private baseUrl = 'http://localhost:8082/accounts';

  constructor(private http: HttpClient) {}

  create(a: Account): Observable<Account> {
    return this.http.post<Account>(this.baseUrl, a);
  }

  getAll(): Observable<Account[]> {
    return this.http.get<Account[]>(this.baseUrl);
  }

  delete(id: number): Observable<string> {
    return this.http.delete<string>(`${this.baseUrl}/${id}`);
  }
}

File: org.springframework.ide.eclipse.boot.dash/account-service/src/main/java/com/bank/account/service/AccountService.java
package com.bank.account.service;

import com.bank.account.entity.Account;
import java.util.List;

public interface AccountService {
    Account create(Account account);
    List<Account> getAll();
    Account getById(Long id);
    Account update(Long id, Account account);
    void delete(Long id);
}
